core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -0,0 +1,358 @@
+                to update.
+            exp_rights_model: ExplorationRightsmodel. The exploration rights
+                model used to update the exploration summary.
+
+        Returns:
+            ExpSummaryModel. The updated exploration summary model to put into
+            the datastore.
+        """
+        exp_summary = exp_services.compute_summary_of_exploration(
+            migrated_exp, exp_rights_model, exp_summary_model)
+        exp_summary.version += 1
+        updated_exp_summary_model = (
+            exp_services.populate_exp_summary_model_fields(
+                exp_summary_model, exp_summary
+            )
+        )
+
+        return updated_exp_summary_model
+
+    @staticmethod
+    def _update_exploration(
+        exp_model: exp_models.ExplorationModel,
+        exp_rights_model: exp_models.ExplorationRightsModel,
+        migrated_exp: exp_domain.Exploration,
+        exp_changes: Sequence[exp_domain.ExplorationChange]
+    ) -> Sequence[base_models.BaseModel]:
+        """Generates newly updated exploration models.
+
+        Args:
+            exp_model: ExplorationModel. The exploration which should be
+                updated.
+            exp_rights_model: ExplorationRightsModel. The exploration rights
+                model which is to be updated.
+            migrated_exp: Exploration. The migrated exploration domain
+                object.
+            exp_changes: sequence(ExplorationChange). The exploration changes
+                to apply.
+
+        Returns:
+            sequence(BaseModel). Sequence of models which should be put into
+            the datastore.
+        """
+        updated_exp_model = (
+            exp_services.populate_exp_model_fields(
+                exp_model, migrated_exp))
+
+        commit_message = (
+            'Update exploration states schema version to %d.'
+        ) % (
+            feconf.CURRENT_STATE_SCHEMA_VERSION
+        )
+        change_dicts = [change.to_dict() for change in exp_changes]
+        with datastore_services.get_ndb_context():
+            if exp_services.get_story_id_linked_to_exploration(
+                migrated_exp.id) is not None:
+                exp_services.validate_exploration_for_story(migrated_exp, True)
+
+            models_to_put = updated_exp_model.compute_models_to_commit(
+                feconf.MIGRATION_BOT_USERNAME,
+                feconf.COMMIT_TYPE_EDIT,
+                commit_message,
+                change_dicts,
+                additional_models={'rights_model': exp_rights_model}
+            ).values()
+
+            exp_services.discard_draft(
+                exp_model.id, feconf.MIGRATION_BOT_USERNAME)
+
+            if opportunity_services.is_exploration_available_for_contribution(
+                migrated_exp.id):
+                (
+                    opportunity_services
+                    .update_opportunity_with_updated_exploration(
+                        migrated_exp.id))
```

Comment:
This should be handled using Beam code. Basically rewrite `is_exploration_available_for_contribution` and `update_opportunity_with_updated_exploration` to Beam operations.

Reply:
Implemented these as beam operations. PTAL.